packages/mui-material/src/Alert/Alert.js

```diff
@@ -167,11 +168,33 @@ const Alert = React.forwardRef(function Alert(inProps, ref) {
 
   const classes = useUtilityClasses(ownerState);
 
-  const AlertCloseButton = slots.closeButton ?? components.CloseButton ?? IconButton;
-  const AlertCloseIcon = slots.closeIcon ?? components.CloseIcon ?? CloseIcon;
+  const backwardCompatibleSlots = {
+    closeButton: components.CloseButton,
+    closeIcon: components.CloseIcon,
+    ...slots,
+  };
+  const backwardCompatibleSlotProps = {
+    ...componentsProps,
+    ...slotProps,
+  };
 
-  const closeButtonProps = slotProps.closeButton ?? componentsProps.closeButton;
-  const closeIconProps = slotProps.closeIcon ?? componentsProps.closeIcon;
+  const [CloseButtonSlot, closeButtonProps] = useSlot('closeButton', {
+    elementType: IconButton,
+    externalForwardedProps: {
+      slots: backwardCompatibleSlots,
+      slotProps: backwardCompatibleSlotProps,
+    },
```

Comment:
What do you think about moving `externalForwardedProps` above?

```js
const externalForwardedProps = {
    slots: {
      closeButton: components.CloseButton,
      closeIcon: components.CloseIcon,
      ...slots,
   },
   slotProps: {
     ...componentsProps,
     ...slotProps,
   },
}

```

Reply:
Much better https://github.com/mui/material-ui/pull/40681/commits/2a7780b8b36eba73ffbb3c2114ca9904ee5b99c6 🙌🏼